openmc/mesh.py

```diff
@@ -75,8 +75,8 @@ def __eq__(self, mesh2):
             return False
 
         # Check lower left / upper right
-        elif self._lower_left != mesh2.lower_left and \
-             self._upper_right != mesh2.upper_right:
+        elif (self._lower_left != mesh2.lower_left).all() and \
```

Comment:
@nelsonag my understanding is that EqualityMixin has been mostly used in the data submodule, while the array/list comparisons throughout the rest of the API use this more standard approach. Do you believe we should eventually convert everything over to use EqualityMixin? If so, what would be the benefit?